File: March_2022/28_03_2022/Palindrome_and_repdigit.cpp
#include <iostream>
#include <math.h>
#include <string>
#include <set>

using namespace std;

bool is_rep(long long x){
    string n = to_string(x);
    set<char> temp;
    for(int i=0;i<n.length();i++){
        temp.insert(n[i]);
    }
    if(temp.size() == 1){
        return true;
    }
    return false;
}

bool is_not_palindrome(long long i){
    string n = to_string(i);
    for(int i=0;i<n.length();i++){
        if(n[i] != n[n.length()-i-1]){
            return true;
        }
    }
    return false;
}

int main(){

    long long start,end;
    cin>>start>>end;

    long long start_cube = cbrt(start);
    long long end_cube = cbrt(end);
    for(int i=10;i<end_cube+1;i++){
        if(is_not_palindrome(pow(i,3)) && is_rep(i)){
            printf("%d\t%d\n",i*i*i,i);
        }
    }

    return 0;
}

File: March_2022/28_03_2022/Palindrome_and_repdigit.py
def isrep(n):
    n = str(n)
    for i in range(len(n)-1):
        if(n[i] != n[i+1]):
            return False
    return True

start = int(input())
end = int(input())
start_cube = int(start**(1/3))
end_cube = int(end**(1/3))

for i in range(10,end_cube+1):
    if(str(i**3) != str(i**3)[::-1]):
        if(isrep(i)):
            print(i**3,i,sep="\t")